src/common/src/config.rs

```diff
@@ -503,6 +503,10 @@ pub struct StorageConfig {
     #[serde(default)]
     pub meta_cache_capacity_mb: Option<usize>,
 
+    /// max memory usage for large query
+    #[serde(default)]
+    pub large_query_memory_usage_mb: Option<usize>,
```

Comment:
After an offline discussion with @fuyufjh,  it was recommended to remove the limitation `large_query_memory_usage_mb`. This is because the memory consumption of streaming_read should be considered as system memory and not handled internally by storage.

Moreover, it is important to note that streaming_read has already been specified on the caller side, so it is expected that the caller follows the API contract. Failing to do so may result in unpredictable system performance.

Reply:
I will set it a large value by default for better performance. But it is still important to keep this variable because we need some way to recover from OOM.